### PR TITLE
Added timeout to `w.clusters.ensure_cluster_running()`

### PR DESCRIPTION
**Pull request by judahrand**
## Changes

This picks up where #118 left off.

## Tests

- [x] `make test` run locally
- [x] `make fmt` applied
- [x] relevant integration tests applied

### CODE REVIEW

File: databricks/sdk/mixins/compute.py [hunk 1]
```diff
@@ -203,16 +209,25 @@ def select_node_type(self,
             return nt.node_type_id
         raise ValueError("cannot determine smallest node type")
 
-    def ensure_cluster_is_running(self, cluster_id: str):
+    def ensure_cluster_is_running(self, cluster_id: str) -> compute.ClusterDetails:
```
Comment (nfx) [inline databricks/sdk/mixins/compute.py:212]: why do you need cluster details to be returned? this will have to be adjusted across all SDKs

Comment (judahrand) [reply databricks/sdk/mixins/compute.py:212]: It doesn't have to 🤷‍♂️ Just seemed more useful given that we have it than returning `None`. 

Do you want me to revert that commit?

Comment (nfx) [reply databricks/sdk/mixins/compute.py:212]: For now, yes.

Comment (nfx) [reply databricks/sdk/mixins/compute.py:212]: @judahrand could you address this comment?

Comment (judahrand) [reply databricks/sdk/mixins/compute.py]: Sorry! I'd forgotten to follow up on this. Done now.

**Review by nfx — APPROVED** [review]

**Pull request merged**